monitoring/api/MonitoringTest/MonitoringTest.cs

```diff
@@ -12,22 +12,36 @@
 // License for the specific language governing permissions and limitations under
 // the License.
 
-using System.IO;
-using Xunit;
+using Grpc.Core;
 using System;
 using System.Linq;
-using System.Net;
-using Grpc.Core;
+using Xunit;
 
 namespace GoogleCloudSamples
 {
     public class TestFixture : IDisposable
     {
+        private const string TimeSeriesQuotaError = "One or more TimeSeries could not be written:" +
+            " One or more points were written more frequently than the maximum sampling period configured for the metric.";
+        public static RetryRobot TimeSeriesQuota { get; } = new RetryRobot
+        {
+            ShouldRetry = ex =>
+                (ex is RpcException rpcEx

```

Comment:
Possibly extract the first two expression into method, so you can have

```csharp
ShouldRetry = ex =>
    IsTimeSeriesQuotaError(ex) ||
    ex is AggregateException aggEx && aggEx.Any(IsTimeSeriesQuotaError);
```

I think that would be clearer - assuming I've read the code correctly.

Reply:
Ah yes, way better, done.